### DISCUSSION TITLE
Ipsec like VPN GATEWAY.

### DISCUSSION THREAD

**Discussion by sanek190** [strongswan/strongswan#1899] [General]
I have a server connected to a VPN (referred to as VPN GW), and I want to use it as a VPN gateway for all the other servers on the network. 
The configuration looks like this. IP addresses are changed for example purposes, and I haven't specified the PSK.

      local_addrs  = 192.168.10.26
      remote_addrs = 10.10.10.10
      rekey_time = 8640
      proposals = aes256-sha256-modp2048
      version = 2
      mobike = no

      local {
         auth = psk
         id = vpn.test
      }
      remote {
         auth = psk
         id = 10.10.10.10
      }
      children {

         local-net1 {
           local_ts  = 192.168.10.29/32
            remote_ts = 10.10.113.7/32
            start_action = trap|start
            dpd_action = restart
            rekey_time = 28800
            mode = tunnel
            esp_proposals = aes256-sha256-modp2048

Now, as a verification step on the VPN GW server, I create a rule:

`iptables -t nat -D POSTROUTING -s 192.168.10.26/32 -d 10.10.113.7/32 -o eth0 -j SNAT --to-source 192.168.10.29`

In this case, the ping from VPN GW successfully reaches the remote_ts. 

Now, on the server 192.168.10.29, which I also want to route traffic to remote_ts through VPN GW, I add a route:

`ip route add 10.10.113.7/32 via 192.168.10.26 dev eth0`

Then the ping from 192.168.10.29 doesn't reach 10.10.113.7. Here is what the table 220 looks like on VPN GW:

`10.10.113.7 via 192.168.10.1 dev eth0 proto static src 192.168.10.26`

"swanctl --list-sas" shows packets in/out but there is no response from ping.

The only solution for ping to work is to add a rule on the server 192.168.10.29:

`iptables -t nat -A OUTPUT  -d 10.10.113.7  -j DNAT --to-destination 192.168.10.26`

And on the VPN GW server:

`iptables -t nat -A PREROUTING -s 192.168.10.29 -j DNAT --to-destination 10.10.113.7`

Why aren't the routes from the client to the VPN GW working?

**Comment by tobiasbrunner** [comment]
Routes != IPsec policies (traffic selectors), maybe read [the introduction](https://docs.strongswan.org/docs/5.9/howtos/introduction.html).

**Reply by sanek190** [reply]
Can I somehow attach an IPsec policy to the packets specified in my route through a VPN GW on the client side?

**Reply by tobiasbrunner** [reply]
That would get dropped by the policy on the other end.  But I suggest you read up on policy- vs. route-based IPsec.
